core/src/components/popover/test/basic/popover.e2e.ts

```diff
@@ -235,3 +150,25 @@ describe('popover: focus trap', () => {
     expect(scrollTop).toBeGreaterThanOrEqual(previousScrollTop);
   });
 });
+
+/**
+ * Focusing happens async inside of popover so we need
+ * to wait for the requestAnimationFrame to fire.
+ */
+const expectActiveElementTextToEqual = async (page: E2EPage, textValue: string) => {
+  await page.evaluate((text) => document.activeElement!.textContent === text, textValue);
```

Comment:
Are you able to use `toBeFocused` instead? https://playwright.dev/docs/test-assertions#locator-assertions-to-be-focused

I had a few `getActiveElement` utilities with the modal tests that I migrated to this solution: https://github.com/ionic-team/ionic-framework/blob/main/core/src/components/modal/test/basic/modal.e2e.ts#L25

Reply:
Also same question about the other activeElement utils

Reply:
To circle back from Slack: we decided to leave this as-is since the change was causing mysterious failures. I added a tech debt ticket with more info.